### PR TITLE
v1.0 rc

### LINKED ISSUE #28: Auro Toast final release
# Publish Component

## Specific steps required to publish this component.

1. __Design & Planning__
    - [x] Reviewed [AuroLabs documentation](http://auro.alaskaair.com/)
    - [x] Reviewed [Auro's Contributing Guidelines](http://auro.alaskaair.com/contributing)
    - [x] Reviewed [Auro's CSS conventions](http://auro.alaskaair.com/webcorestylesheets/conventions) and how to apply CSS to custom elements
    - [x] Design team engaged in designing the new component
    - [x] Design work integrated in the Auro Design System UI Kit
1. __Development__
    - [x] Addresses all design and UX as proposed in the design comp
    - [x] A11y in regards to role and live notifications need to be coded into the final WC
    - [x] All style code follows [Auro's Style Guide](http://auro.alaskaair.com/webcorestylesheets/guidelines)
    - [x] Automated tests created
1. __Documentation__
    - `demo.md`
      - [x] Component properly described
      - [x] Custom element's full API is documented
      - [x] Component use case(s) outlined
      - [x] `Do Not` section outlines implementation scenarios to avoid
      - Example Section
        - [x] Basic example of simplest use case
        - [x] Examples showing use of each API attribute
        - [x] All examples include rendered example and matching code sample
    - `readme.md`
      - [x] Component properly described
      - [x] Bundled version info
      - [x] Use cases described
      - [x] `API Code Examples` updated to match API with working example code
    - [x] `API.md` document created and matches current API
1. __Testing__
    - [x] Component built using latest version of [WC-Generator](http://auro.alaskaair.com/getting-started/developers/generator/install)
    - [x] All automated tests pass and coverage threshold requirements are met
    - [x] [Browser compatibility](http://auro.alaskaair.com/support/browsersSupport) - desktop and mobile versions
    - [Accessibility](http://auro.alaskaair.com/a11y-statement) - One or more of the following
      - [x] Axe / Google Lighthouse (baseline standard)
      - [x] Screenreader (determine usability of experience via screenreader)
      - [x] Keyboard use (a user must be able to navigate to key parts via a keyboard)
      - [x] Accessibility should be verified against all browsers
    - [x] Bundled versions tested
1. __Review__ <br/>
    When the component is ready for a final review it is the responsibility of the contributor to coordinate the following reviews:
    - [x] Reviewed demo with the designer
    - [x] Reviewed demo with Auro Team
    - [ ] All work is reviewed by Auro team
1. __Publishing__
    - [x] Removed all references to `auroLabs` from the repository
    - [x] Component properly named
    - [x] GitHub Repository properly named
    - Updated `readme.md`
      - [x] Install information updated to published namespace
      - [x] `Define dependency in project component` updated to published namespace
      - [x] `Bundle example code` updated to published namespace
    - [ ] PR prepared to include demo page in the Auro Doc Site

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is, and anything that describes the before/after state of the project._

This is the summary PR for releasing the v1.0 version of auro-toast. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -3,13 +3,13 @@ repository:
   # See https://github.com/apps/settings for app settings
 
   # A short description of the repository that will show up on GitHub
-  description: Placeholder repo
+  description: Custom element that ....
```
Comment (cynthiaricomendoza-alaskaair) [inline .github/settings.yml]:
```suggestion
  description: Custom element that illustrates a quick message after an action is complete
```

**Review by cynthiaricomendoza-alaskaair — DISMISSED** [review]
lgtm - thanks for making those editorial changes!

File: scripts/generateDocs.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+
+  /**
+   * Cleanup line breaks
+   */
+  result = result.replace(/(\r\n|\r|\n)[\s]+(\r\n|\r|\n)/g, '\r\n\r\n'); // Replace lines containing only whitespace with a carriage return.
+  result = result.replace(/>(\r\n|\r|\n){2,}/g, '>\r\n'); // Remove empty lines directly after a closing html tag.
+  result = result.replace(/>(\r\n|\r|\n)```/g, '>\r\n\r\n```'); // Ensure an empty line before code samples.
+  result = result.replace(/>(\r\n|\r|\n){2,}```(\r\n|\r|\n)/g, '>\r\n```\r\n'); // Ensure no empty lines before close of code sample.
+  result = result.replace(/([^(\r\n|\r|\n)])(\r\n|\r|\n)+#/g, "$1\r\n\r\n#"); // Ensure empty line before header sections.
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply scripts/generateDocs.js:63]: The following PR is being worked on to address this issue 

- https://github.com/AlaskaAirlines/WC-Generator/pull/410

**Comment by blackfalcon** [comment]
Chrome mobile experience 

Screen Shot 2023-07-10 at 3 21 02 PM

Safari 

Screen Shot 2023-07-10 at 3 22 32 PM

Firefox

Screen Shot 2023-07-10 at 3 23 18 PM

**Comment by blackfalcon** [comment]
Lighthouse A11Y score. Issue is NOT related to the component, has to do with the coloring of the code examples. 

Screen Shot 2023-07-10 at 3 24 55 PM

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: "CodeQL"
```
Comment (jordanjones243) [inline .github/workflows/codeql.yml:1]: Missing `autoAssign.yml`

File: demo/addToast.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+// ---------------------------------------------------------------------
+
+showToast = (toastID) => {
```
Comment (jordanjones243) [inline demo/addToast.js]: This should be located under `apiExamples/` and should be ran under a file that runs all javascript examples for each respective page, like so: https://github.com/AlaskaAirlines/auro-combobox/blob/main/demo/index.js

File: rollup.config.mjs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    nodeResolve(),
+    !production &&
+      serve({
+        open: true,
+        openPage: '/docs/'
+      })
+  ]
+};
+
+export default [modernConfig];
```
Comment (jordanjones243) [inline rollup.config.mjs]: Need to update rollup.config: https://github.com/AlaskaAirlines/auro-combobox/blob/main/rollup.config.mjs

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Need to run a build to update docs, an event type was changed.

File: demo/addToast.js (hunk 4 above)
Comment (blackfalcon) [reply demo/addToast.js]: @jordanjones243 please consult with @cynthiaricomendoza-alaskaair to assist with this update. This follows the new example.js model that I am not all that failure with.

File: rollup.config.mjs (hunk 5 above)
Comment (blackfalcon) [reply rollup.config.mjs]: @jordanjones243 again, please follow up with @cynthiaricomendoza-alaskaair on this. 

Also, this model is not part of the generator. I was unaware that this was a pattern that we are actively doing with all components. 

@Patrick-Daly-AA there should be an issue created with the WC-Generator to add this functionality. Please work with @jordanjones243 to define that specification.

File: .github/workflows/codeql.yml (hunk 3 above)
Comment (blackfalcon) [reply .github/workflows/codeql.yml:1]: Added to the repo.

File: demo/index.html [hunk 6]
```diff
@@ -43,13 +43,16 @@
...
+    <script src="https://cdn.jsdelivr.net/npm/@alaskaairux/auro-button@latest/dist/auro-button__bundled.js" type="module"></script>
     <script type="module" src="../index.js"></script>
     <script type="module">
       import { registerComponent } from "../index.js"
       registerComponent('custom-toast');
     </script>
     <script type="module" src="../src/auro-toaster.js"></script>
-    <script src="./addToast.js"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-button@latest/dist/auro-button__bundled.js" type="module"></script>
+    <script type="module" src="./index.min.js"></script>
```
Comment (jordanjones243) [inline demo/index.html]: type="module" needs to be removed from this line and the one below

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: demo/index.html (hunk 6 above)
Comment (cynthiaricomendoza-alaskaair) [reply demo/index.html]: done! all pushed up

File: demo/addToast.js (hunk 4 above)
Comment (cynthiaricomendoza-alaskaair) [reply demo/addToast.js]: met with Jordan yesterday and pushed up changes for this!

File: rollup.config.mjs (hunk 5 above)
Comment (cynthiaricomendoza-alaskaair) [reply rollup.config.mjs]: met with Jordan yesterday and pushed up changes for this!

File: demo/index.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { showToast } from '../apiExamples/showToast';
+
+export function initToastIndexExamples(initCount) {
+  initCount = initCount || 0;
+
+  try {
+    showToast();
```
Comment (jordanjones243) [inline demo/index.js:8]: Comment this line out as it is no longer needed and leave comment above this line saying "add new javascript function calls to test examples on index page"

Do the same thing for the api page

File: docs/partials/apiExamples.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+<div class="exampleWrapper">
+  <!-- AURO-GENERATED-CONTENT:START (FILE:src=./../../apiExamples/visible.html) -->
+  <!-- AURO-GENERATED-CONTENT:END -->
+</div>
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+```js
+// Example function
```
Comment (jordanjones243) [inline docs/partials/apiExamples.md]: This needs to be updated to auto generate the script with `apiExamples/showToast.js` as you see on line 13 and 14. There needs to be documentation explaining that every button is using this file.

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: docs/partials/apiExamples.md (hunk 8 above)
Comment (cynthiaricomendoza-alaskaair) [reply docs/partials/apiExamples.md]: update in this commit: https://github.com/AlaskaAirlines/auro-toast/pull/27/commits/4dd55edb65c0ac21070401e7ce8c9c5f295cad6b

File: demo/index.js (hunk 7 above)
Comment (cynthiaricomendoza-alaskaair) [reply demo/index.js:8]: i added a comment, but kept showToast() since that is needed in order for the button to work.

**Review by jordanjones243 — APPROVED** [review]

**Pull request merged**